opencog/atoms/base/Atom.cc

```diff
@@ -201,6 +201,17 @@ void Atom::setAttentionValue(AttentionValuePtr av)
     avch(getHandle(), local, av);
 }
 
+void Atom::stimulate(AtomSpace* as, float stimulus)
```

Comment:
The solution that makes sense is to add a method to AtomSpace something like 
```
AtomSpace::stimulate(const Handle& h, double stimulus)
```
that does what your code does, but flipping things around.

Also -- please don't use float -- use double.  floats are just 32 bit, with 23-bit mantissas and will plague you with rounding errors.